### PR TITLE
Try reenabling helix for components

### PR DESCRIPTION
**Pull request by HaoK**

### CODE REVIEW

**Comment by HaoK** [comment]
@MattGal looks like our "--" might have broken the attachment in the work items for failing tests to the download, it always goes to an empty tab for me now :(

I thought java JDK was already installed in the past, at least in the comments of https://github.com/dotnet/core-eng/issues/5302

I assumed it would already be on the path for the helix work items, but I'm seeing java not found (when trying to install things from npm/yarn)

```
Copying C:\Users\runner\AppData\Local\Temp\nodejs\node-v10.15.3-win-x64\* to .
Node.exe copied to .
C:\h\w\A5E808F7\w\AEE809F1\e\yarn -> C:\h\w\A5E808F7\w\AEE809F1\e\node_modules\yarn\bin\yarn.js
C:\h\w\A5E808F7\w\AEE809F1\e\yarnpkg -> C:\h\w\A5E808F7\w\AEE809F1\e\node_modules\yarn\bin\yarn.js
+ yarn@1.21.1
added 1 package in 0.594s
yarn install v1.21.1
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
$ selenium-standalone install
not found: java
Usage: selenium-standalone action [options]
error Command failed with exit code 255.

```

**Comment by HaoK** [comment]
@javiercn can you help me figure out how to get yarn working?  Looks like its not a java on path issues since I manually downloaded and installed it, and yarn still fails to install selenium like so:

```
C:\h\w\989D0856\w\973D0894\e\yarnpkg -> C:\h\w\989D0856\w\973D0894\e\node_modules\yarn\bin\yarn.js
C:\h\w\989D0856\w\973D0894\e\yarn -> C:\h\w\989D0856\w\973D0894\e\node_modules\yarn\bin\yarn.js
+ yarn@1.21.1
added 1 package in 0.563s
yarn install v1.21.1
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
$ selenium-standalone install
not found: java
Usage: selenium-standalone action [options]
error Command failed with exit code 255.

"Java?"
Usage: java [options] <mainclass> [args...]
           (to execute a class)
   or  java [options] -jar <jarfile> [args...]
           (to execute a jar file)
   or  java [options] -m <module>[/<mainclass>] [args...]
       java [options] --module <module>[/<mainclass>] [args...]
           (to execute the main class in a module)
   or  java [options] <sourcefile> [args]
           (to execute a single source-file program)



```

**Comment by javiercn** [comment]
> can you help me figure out how to get yarn working? Looks like its not a java on path issues since I manually downloaded and installed it, and yarn still fails to install selenium like so:

I'm a bit confused, isn't that exactly what the build is complaining about? If you do java -v in the context where the build is run, does it succeed?

**Comment by HaoK** [comment]
Oh i think I see the issue, the path isn't setup early enough

File: eng/targets/Helix.props [hunk 1]
```diff
@@ -44,4 +45,9 @@
     <HelixPreCommand Include="call RunPowershell.cmd InstallNode.ps1 $(NodeVersion) %25HELIX_CORRELATION_PAYLOAD%25\node\bin || exit /b 1" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TestDependsOnYarn)' == 'true' AND '$(IsWindowsHelixQueue)' == 'true'">
+    <HelixPreCommand Include="call RunPowershell.cmd InstallJdk.ps1 $(JdkVersion) || exit /b 1" />
```
Comment (BrennanConroy) [inline eng/targets/Helix.props]: Split these into two different groups. One for yarn and one for java.

Comment (HaoK) [reply eng/targets/Helix.props]: Good idea

**Comment by HaoK** [comment]
@javiercn getting closer, where's this config file coming from for the selenium, need to change where its looking on helix

```
yarn install v1.21.1
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
$ selenium-standalone install --config ../../../Shared/E2ETesting/selenium-config.json
Error parsing config file : Cannot find module 'C:\h\w\AC56097A\Shared\E2ETesting\selenium-config.json'
Usage: selenium-standalone action [options]
error Command failed with exit code 255.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

```

**Comment by HaoK** [comment]
@pranavkm looks like https://github.com/dotnet/aspnetcore/pull/18676/files change is making this harder for components to run on helix since its now looking for a relative path, can we do something to make this look for that config file in the local directory (just copy it over as part of build or something?)

**Comment by HaoK** [comment]
@javiercn looks like the content root being generated at compile time is not going to work well on helix, can you look into a fix for this to work on helix, we control how the app is packaged and deployed so a relative path from the current directory should work I think?


https://github.com/dotnet/aspnetcore/blob/e72223eaf58a3ee6660a922064d2449e47b78253/src/Components/test/testassets/TestServer/Components.TestServer.csproj#L28

The error at runtime is a directory not found:

```
  Error Message:
   System.IO.DirectoryNotFoundException : F:\workspace\_work\1\s\src\Components\test\testassets\TestServer\
  Stack Trace:
     at Microsoft.Extensions.FileProviders.PhysicalFileProvider..ctor(String root, ExclusionFilters filters)
   at Microsoft.Extensions.FileProviders.PhysicalFileProvider..ctor(String root)
   at Microsoft.Extensions.Hosting.HostBuilder.CreateHostingEnvironment()
   at Microsoft.Extensions.Hosting.HostBuilder.Build()
   at TestServer.Program.BuildWebHost[TStartup](String[] args) in F:\workspace\_work\1\s\src\Components\test\testassets\TestServer\Program.cs:line 68
   at Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures.AspNetSiteServerFixture.CreateWebHost() in /_/src/Components/test/E2ETest/Infrastructure/ServerFixtures/AspNetSiteServerFixture.cs:line 37
   at Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures.WebHostServerFixture.StartAndGetRootUri() in /_/src/Components/test/E2ETest/Infrastructure/ServerFixtures/WebHostServerFixture.cs:line 17
   at Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures.ServerFixture.<.ctor>b__4_0() in /_/src/Components/test/E2ETest/Infrastructure/ServerFixtures/ServerFixture.cs:line 25

```

**Comment by javiercn** [comment]
> @javiercn looks like the content root being generated at compile time is not going to work well on helix, can you look into a fix for this to work on helix, we control how the app is packaged and deployed so a relative path from the current directory should work I think?

I think so, you could key that of from `IsHelixBuild`?

**Comment by HaoK** [comment]
Okay finally some progress, @javiercn can you see if you can make out what's actually failing now?

https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-aspnetcore-refs-pull-18636-merge-76431147d907455ca6/Microsoft.AspNetCore.Components.E2ETests--netcoreapp5.0/console.99371bcb.log?sv=2019-02-02&se=2020-02-15T18%3A17%3A29Z&sr=c&sp=rl&sig=qB0WLFnBR4PsbOvMNGzwYWB3z8yad8QQm47i22BvTEo%3D

I'm seeing some errors around launching the server it looks like

**Comment by javiercn** [comment]
It seems is null refing here?

https://github.com/dotnet/aspnetcore/blob/master/src/Shared/E2ETesting/SeleniumStandaloneServer.cs#L202-L215

Not sure what can be causing it to fail there.

**Comment by javiercn** [comment]
@HaoK are the agents persistent or are they transient, in the sense they get re-imagined/rebooted per run.

We could try and remove the process tracking logic.

**Comment by HaoK** [comment]
@javiercn yeah i don't think we need process tracking logic at all on the helix agents, they shouldn't be getting reused

**Comment by javiercn** [comment]
> @javiercn yeah i don't think we need process tracking logic at all on the helix agents, they shouldn't be getting reused

I suggest we disable it then

**Comment by HaoK** [comment]
Sounds good to me, can you just push onto this branch the removal of the process tracking stuff that you don't think we need?

**Comment by javiercn** [comment]
@HaoK Is there a Helix env variable I can key this off?

**Comment by HaoK** [comment]
Yeah just use
```
 if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("helix")))
```

**Comment by HaoK** [comment]
branch looks messed up with builds not triggering, doing a merge/force push here @javiercn , just a heads up in case you have changes locally

**Comment by HaoK** [comment]
Progress I think, tests are still failing, console output is here @javiercn :

  https://helix.dot.net/api/2019-06-17/jobs/d42b348b-03ce-4a9c-88f6-086fa01d7fcb/workitems/Microsoft.AspNetCore.Components.E2ETests--netcoreapp5.0/files/console.e25e0e27.log

**Comment by MattGal** [comment]
@HaoK I'm unsubscribing from this thread as it's a bit noisy, but if you need my input on anything just let me know directly please.

**Comment by HaoK** [comment]
@javiercn @pranavkm so it looks like installing yarn nukes npm, which then causes issues when selenium runs, any ideas how these two are supposedly to play together, what should the steps be on the helix side for things to be installed correctly?  Right now I just tried using npm to install yarn, and then running yarn install, which looks like it works, but when the tests run selenium blows up because npm's cli gets nuked by yarn I think

**Comment by pranavkm** [comment]
@haok, you should be able to switch the test runner to use `yarn` instead of `npm`.  Basically replace `npm` -> `yarn` here: 

* https://github.com/dotnet/aspnetcore/pull/18636/files#diff-d212f4bd0f7a304eb6bcc26b1802b69aR101
* https://github.com/dotnet/aspnetcore/pull/18636/files#diff-d212f4bd0f7a304eb6bcc26b1802b69aR110

**Comment by HaoK** [comment]
@pranavkm so looks like both yarn and npm are missing from node_modules after yarn install,

 https://helix.dot.net/api/2019-06-17/jobs/877f305e-496c-4aea-a656-aec446c4ac16/workitems/Microsoft.AspNetCore.Components.E2ETests--netcoreapp5.0/files/console.37ab6af8.log

Any idea what nukes those?

**Comment by HaoK** [comment]
Okay I think I got selenium working now, but the tests appear to be hanging, @pranavkm @javiercn can you guys tell what's going on from the exceptions, the screenshots are taken but its hanging so they don't get uploaded I think, I can try having the test directly write to the upload directory

https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-aspnetcore-refs-pull-18636-merge-6b03aadab2d64a749c/Microsoft.AspNetCore.Components.E2ETests--netcoreapp5.0/console.c865d749.log?sv=2019-02-02&se=2020-03-24T09%3A01%3A48Z&sr=c&sp=rl&sig=zt9RkAPid99FI6EIiNycGKQ6w5AJhHOCNCgYlbia3rk%3D

**Comment by HaoK** [comment]
Replaced by https://github.com/dotnet/aspnetcore/pull/21280

**Pull request closed without merging**